### PR TITLE
Replace AJAX search with Live component

### PR DESCRIPTION
**Pull request by ker0x**
This pull request proposes to replace the AJAX search with a live component example

### CODE REVIEW

**Comment by 94noni** [comment]
its a nice idea, following stimulus intro 👍🏻 
but it could be wise to also add some code doc to link to documentation etc, as just browsing the demo code without explanation can be disturbing imho

File: templates/blog/search.html.twig [hunk 1]
```diff
@@ -2,24 +2,14 @@
...
-                   class="form-control search-field"
-                   placeholder="{{ 'post.search_for'|trans }}"
-                   autocomplete="off"
-                   value="{{ query }}"
-                   autofocus
-                   data-no-results-message="{{ 'post.search_no_results'|trans }}"
-            >
-        </div>
-    </form>
+    {{ component('blog_search', {
```
Comment (94noni) [inline templates/blog/search.html.twig:10]:
```suggestion
    {# Lear more on https://symfony.com/bundles/ux-live-component/current/index.html #}
    {{ component('blog_search', {
```

File: templates/components/blog_search.html.twig [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<div {{ attributes }}>
```
Comment (94noni) [inline templates/components/blog_search.html.twig:1]: same as above, adding link to `{# Lear more on https://symfony.com/bundles/ux-live-component/current/index.html #}`

File: src/Twig/Components/BlogSearchComponent.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+namespace App\Twig\Components;
+
+use App\Entity\Post;
+use App\Repository\PostRepository;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+
+/**
+ * Live component to display instant search for Posts.
```
Comment (94noni) [inline src/Twig/Components/BlogSearchComponent.php:21]: the file https://github.com/symfony/demo/blob/main/src/Command/AddUserCommand.php describe some usage etc
could be wise to do same here, wdyt?

for ex to https://ux.symfony.com/live-component or/and https://symfony.com/bundles/ux-live-component/current/index.html

Comment (ker0x) [reply src/Twig/Components/BlogSearchComponent.php:21]: Good idea indeed. I added some comments in my code!

File: templates/blog/search.html.twig [hunk 4]
```diff
@@ -2,24 +2,18 @@
 
 {% block javascripts %}
     {{ parent() }}
-    {{ encore_entry_script_tags('search') }}
 {% endblock %}
```
Comment (rosier) [inline templates/blog/search.html.twig:5]: This javascripts block can be remove now

Comment (ker0x) [reply templates/blog/search.html.twig]: Indeed, nice catch thanks!

**Review by 94noni — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Romain, thanks a lot for contributing this nice feature. I feared that introducing an UX example would complicate things too much ... but making this change was simply perfect because the result is very nice and it allows to compare traditional JS features vs modern UX-based JS-like features.

Thanks to reviewers too 🙏

**Comment by COil** [comment]
Excellent, that's the kind of "real use case" that really shows the power of UX. 👏

**Pull request merged**